source/fundamentals/typescript.txt

```diff
@@ -79,81 +80,51 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
 class in the
 :ref:`Find Multiple Documents Usage Example <node-driver-find-usage-example-code-snippet>`.
 
+.. _node-driver-limitations:
+
 Limitations
 -----------
 
-.. _node-driver-typescript-limitations-dot-notation:
+.. _node-driver-recursive-type:
 
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
+You cannot specify a recursive type as a type parameter for {+driver-long+}
+classes. Click the following tabs to see a recursive type and a code snippet that
+uses the recursive type and raises an error: 
 
 .. tabs::
 
-  .. tab:: Dot Notation
-     :tabid: dot-notation
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
-
-     This is the error:
+   .. tab:: Recursive Type
+      :tabid: recursive-type
```

Comment:
Suggestion: I wonder if it could be useful to add a sentence that explains when you might use a recursive type in MongoDB and for the example to demonstrate that.

Reply:
I'm not sure I agree with this suggestion. My reasons are as follows:

- You would use a recursive type for a collection that contains a recursive structure (such as a phone tree). I'm not sure describing the real world example would contribute to the message of the section (don't use a recursive type), but would increase the length (and most likely complexity) of the page.
- This page is for [Lucas level readers](https://github.com/mongodb/docs-node/pull/247) who are familiar with recursion and recursive data structures. I'm not sure a real world example will provide much value for these readers, as they likely are able to relate recursive data structures to real world examples independently.